webapp/channels/src/components/admin_console/schema_admin_settings.tsx

```diff
@@ -716,65 +771,56 @@ export default class SchemaAdminSettings extends React.PureComponent {
                 id={setting.key}
                 label={this.renderLabel(setting)}
-                helpText={this.renderHelpText(setting)}
-                placeholder={Utils.localizeMessage(setting.placeholder, setting.placeholder_default) || Utils.localizeMessage('search_bar.search', 'Search')}
+                helpText={this.renderSettingHelpText(setting)}
+                placeholder={setting.placeholder_message}
                 value={this.state[setting.key] || ''}
                 disabled={this.isDisabled(setting)}
                 onChange={this.handleChange}
             />
         );
     };
 
-    buildJobsTableSetting = (setting) => {
+    buildJobsTableSetting = (setting: AdminDefinitionSetting) => {
+        if (!this.props.schema || setting.type !== 'jobstable') {
+            return (<></>);
+        }
+
         return (
             <JobsTable
                 key={this.props.schema.id + '_jobstable_' + setting.key}
                 jobType={setting.job_type}
                 getExtraInfoText={setting.render_job}
                 disabled={this.isDisabled(setting)}
-                createJobButtonText={
-                    <FormattedMessage
-                        id={setting.label}
-                        defaultMessage={setting.label_default}
-                    />
-                }
-                createJobHelpText={
-                    <FormattedMarkdownMessage
-                        id={setting.help_text}
-                        defaultMessage={setting.help_text_default}
-                    />
-                }
+                createJobButtonText={descriptorOrStringToString(setting.label, this.props.intl)}
+                createJobHelpText={this.renderSettingHelpText(setting)}
             />
         );
     };
 
-    buildFileUploadSetting = (setting) => {
-        const setData = (id, data) => {
-            const successCallback = (filename) => {
-                this.handleChange(id, filename);
-                this.setState({[setting.key]: filename, [`${setting.key}Error`]: null});
-            };
-            const errorCallback = (error) => {
-                this.setState({[setting.key]: null, [`${setting.key}Error`]: error.message});
-            };
-            setting.set_action(successCallback, errorCallback, data);
-        };
+    buildFileUploadSetting = (setting: AdminDefinitionSetting) => {
+        if (!this.props.schema || setting.type !== 'fileupload' || !setting.key) {
+            return (<></>);
+        }
 
         if (this.state[setting.key]) {
-            const removeFile = (id, callback) => {
+            const removeFile = (id: string, callback: () => void) => {
                 const successCallback = () => {
-                    this.handleChange(setting.key, '');
-                    this.setState({[setting.key]: null, [`${setting.key}Error`]: null});
+                    this.handleChange(setting.key!, '');
```

Comment:
Oh, I see. That's really interesting. Since it's an arrow function, I would've expected the value of `setting.key` to be captured when `successCallback` gets defined, but I guess it's only capturing `setting` while the values inside it can still change